test/utils.py

```diff
@@ -5,3 +5,17 @@
 def equal_images(image_1: Image.Image, image_2: Image.Image) -> bool:
     """Returns True when both input images are equal."""
     return (np.asarray(image_1) == np.asarray(image_2)).all()
+
+
+def equal_images_within_margin(image_1: Image.Image, image_2: Image.Image, error_margin: float) -> bool:
+    """Return True when the percentage of differences within each image is bellow the error_margin."""
+    percentage_of_dissimilarity = percentage_false((np.asarray(image_1) == np.asarray(image_2)))
+    return percentage_of_dissimilarity < error_margin
+
+
+def percentage_false(array: np.ndarray) -> float:
```

Comment:
nit:  with [`numpy.typing.NDArray`](https://numpy.org/devdocs/reference/typing.html#numpy.typing.NDArray) this `np.ndarray` type annotation could be updated to the more-specific `NDArray[bool]` .. _if_ i'm understanding the docs correctly.

it's not like we're actually running mypy in CI (like we probably should -- bandwidth issues, i guess).